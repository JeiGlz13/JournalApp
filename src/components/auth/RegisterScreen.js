import React from 'react';
import {Link} from 'react-router-dom'; 
import { useForm } from '../../hooks/useForm';
import validator from 'validator';
import { useDispatch, useSelector } from 'react-redux';
import { removeError, setError } from '../../actions/ui';
import { startRegisterWithEmailPasswordName } from '../../actions/auth';


export const RegisterScreen = () => {
    const dispatch = useDispatch();
    const {msgError} = useSelector(state => state.ui);

    const [formValues, handleInputChange] = useForm({
            name: '',
            email: '',
            password: '',
            password2: ''
        });

        const {name, email, password, password2} = formValues;

        const handleRegister = (e)=>{
            e.preventDefault();
            if(isFormValid()){
                dispatch(startRegisterWithEmailPasswordName(email, password, name));
            }
        }

        const isFormValid = ()=>{
            if(name.trim().length === 0){
                dispatch(setError('Name is required'));
                return false;
            }else if(!validator.isEmail(email)){
                dispatch(setError('Email is not valid'));
                return false;
            }else if(password !== password2 || password.length < 6){
                dispatch(setError('Password should be at least 6 characters'));
                return false;
            }
            dispatch(removeError());
            return true;
        }

    return (
        <div>
            <h3 className = "auth__title">Register</h3>

            <form onSubmit = {handleRegister} className = "animate__animated animate__fadeIn animate-faster auth__form">
                {   msgError &&
                    <div className= "auth__alert-error">
                        {msgError}
                    </div>
                }

                <input className = "auth__input" type = "text" placeholder = "Name" name = "name" value = {name} onChange = {handleInputChange} autoComplete="off" />
                <input className = "auth__input" type = "text" placeholder = "Email" name = "email" value = {email} onChange = {handleInputChange} autoComplete="off" />
                <input className = "auth__input" type = "password" placeholder = "Password" name = "password" value = {password} onChange = {handleInputChange} />
                <input className = "auth__input" type = "password" placeholder = "Confirm Password" name = "password2" value = {password2} onChange = {handleInputChange} />

                <button className = "btn btn-primary btn-block" type = "submit">
                    Login
                </button>
                
                <div className = "auth__social-networks">

                    <Link className = "link" to = "/auth/login"> 
                        Already registered?
                    </Link>
                </div>
            </form>
        </div>
    )
}
