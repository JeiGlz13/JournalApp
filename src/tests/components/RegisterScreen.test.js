import React from 'react';
import { RegisterScreen } from '../../components/auth/RegisterScreen';
import {mount} from 'enzyme';
import { Provider } from 'react-redux';
import {MemoryRouter} from 'react-router-dom';

import configureStore from 'redux-mock-store'; //ES6 modules
import thunk from 'redux-thunk';
import { types } from '../../types/types';

/* jest.mock('../../actions/auth', ()=>({
    startGoogleLogin: jest.fn(),
    startLoginEmailPassword: jest.fn()
})); */

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {
    auth: {},
    ui: {
        loading: false,
        msgError: null
    }
};

let store = mockStore(initState);


describe('Pruebas en <RegisterScreen/>', () => {

    const wrapper = mount(<Provider store = {store}>
        <MemoryRouter>
            <RegisterScreen/>
        </MemoryRouter>
    </Provider>)

    

   test('Debe mostrarse correctamente', () => {
       expect(wrapper).toMatchSnapshot();
   });
    
   test('Debe hacer el dispatch de la accion', () => {
       const emailField =wrapper.find('input[name="email"]');
       emailField.simulate('change', {
           target: {
               value: '',
               name: 'email'
           }
       });

       wrapper.find('form').simulate('submit', {
           preventDefault(){}
       });

       const actions = store.getActions();
      expect(actions[0]).toEqual({
           type: types.uiSetError,
           payload: 'Email is not valid'
       })
   });
   
   test('Debe mostrar la caja de alerta', () => {
    const initState = {
        auth: {},
        ui: {
            loading: false,
            msgError: 'Email no es correcto'
        }
    };
    
    const store = mockStore(initState);

    const wrapper = mount(
    <Provider store = {store}>
        <MemoryRouter>
            <RegisterScreen/>
        </MemoryRouter>
    </Provider>)

    expect(wrapper.find('.auth__alert-error').exists()).toBe(true);
    expect(wrapper.find('.auth__alert-error').text().trim()).toBe(initState.ui.msgError);
   })
   
});
