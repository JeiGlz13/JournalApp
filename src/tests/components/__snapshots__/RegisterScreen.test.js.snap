// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <RegisterScreen/> Debe mostrarse correctamente 1`] = `
<div>
  <h3
    className="auth__title"
  >
    Register
  </h3>
  <form
    className="animate__animated animate__fadeIn animate-faster"
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="auth__input"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value="Jeisson"
    />
    <input
      autoComplete="off"
      className="auth__input"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      type="text"
      value="jeiglz@gmail.com"
    />
    <input
      className="auth__input"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value="123456"
    />
    <input
      className="auth__input"
      name="password2"
      onChange={[Function]}
      placeholder="Confirm Password"
      type="password"
      value="123456"
    />
    <button
      className="btn btn-primary btn-block"
      type="submit"
    >
      Login
    </button>
    <div
      className="auth__social-networks"
    >
      <a
        className="link"
        href="/auth/login"
        onClick={[Function]}
      >
        Already registered?
      </a>
    </div>
  </form>
</div>
`;
